pytest/tests/stress/stress.py

```diff
@@ -296,107 +293,67 @@ def revert_txs():
+                from_ = random.randint(0, len(nodes) - 1)
+            to = random.randint(0, len(nodes) - 1)
+            while from_ == to:
+                to = random.randint(0, len(nodes) - 1)
+            amt = random.randint(0, min_balances[from_])
+            nonce_val, nonce_lock = nonces[from_]
 
-                            if get_balances() == expected_balances:
-                                trace_reverted_txs(last_tx_set, [i, j])
+            hash_, _ = get_recent_hash(nodes[-1], 5)
 
-                            for k in range(j + 1, len(last_tx_set)):
-                                tx = last_tx_set[k]
-                                expected_balances[tx[1]] += tx[4]
-                                expected_balances[tx[2]] -= tx[4]
+            with nonce_lock:
+                tx = sign_payment_tx(nodes[from_].signer_key, 'test%s' % to,
+                                     amt, nonce_val.value,
+                                     base58.b58decode(hash_.encode('utf8')))
 
-                                if get_balances() == expected_balances:
-                                    trace_reverted_txs(last_tx_set, [i, j, k])
+                # Loop trying to send the tx to all the validators, until at least one receives it
+                tx_hash = None
+                for send_attempt in range(SEND_TX_ATTEMPTS):
+                    shuffled_validator_ids = [x for x in validator_ids]
+                    random.shuffle(shuffled_validator_ids)
+                    for validator_id in shuffled_validator_ids:
+                        try:
+                            info = nodes[validator_id].send_tx(tx)
+                            if 'error' in info and info['error']['data'] == 'IsSyncing':
+                                pass
 
-                                expected_balances[tx[1]] -= tx[4]
-                                expected_balances[tx[2]] += tx[4]
+                            elif 'result' in info:
+                                tx_hash = info['result']
+                                break
 
-                            tx = last_tx_set[j]
-                            expected_balances[tx[1]] -= tx[4]
-                            expected_balances[tx[2]] += tx[4]
+                            else:
+                                assert False, info
 
-                        tx = last_tx_set[i]
-                        expected_balances[tx[1]] -= tx[4]
-                        expected_balances[tx[2]] += tx[4]
+                        except (requests.exceptions.ReadTimeout,
+                                requests.exceptions.ConnectionError):
+                            if not network_issues_expected and not nodes[
+                                    validator_id].mess_with:
+                                raise
 
-                    logging.info(
-                        "The latest and greatest stats on successful/failed: %s/%s"
-                        % (good, bad))
```

Comment:
git fails here to display the diff correctly. I removed the highlighed part here. It was bruteforcing up to three txs to find a set that is sufficient to unroll if the balances don't catch up. It has proven itself not particularly useful, but makes the code rather messy.